ionc/ion_timestamp.c

```diff
@@ -930,26 +1031,42 @@ iERR ion_timestamp_equals(const ION_TIMESTAMP *ptime1, const ION_TIMESTAMP *ptim
         goto is_false;
     }
 
-    IONCHECK(ion_timestamp_get_precision(ptime1, &precision1));
-    IONCHECK(ion_timestamp_get_precision(ptime2, &precision2));
-    if (precision1 != precision2) {
-        goto is_false;
-    }
-
-    IONCHECK(ion_timestamp_to_time_t(ptime1, &t1));
-    IONCHECK(ion_timestamp_to_time_t(ptime2, &t2));
-    if (t1 != t2) {
-        goto is_false;
+    if (instant_only) {
+        // Fractional precision does not matter in this case. Trim any trailing zeros from the fractions, then compare.
+        decQuadReduce(&fraction_trimmed1, &ptime1->fraction, pcontext);
+        decQuadReduce(&fraction_trimmed2, &ptime2->fraction, pcontext);
+        ion_decimal_equals(&fraction_trimmed1, &fraction_trimmed2, pcontext, &decResult);
+        if (!decResult) goto is_false;
```

Comment:
avoiding use of {} for simple if statements like this tends to cause more problems than it solves;  my preference is to always use {}